Devscord.DiscordFramework/Services/CyclicCacheGenerator.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Devscord.DiscordFramework.Services
+{
+    public abstract class CyclicCacheGenerator
+    {
+        private CancellationTokenSource _cancellationTokenSource;
+
+        public async void StartGeneratingStatsCacheEveryday()
+        {
+            var isNowRunning = _cancellationTokenSource?.IsCancellationRequested == false;
+            if (isNowRunning)
+            {
+                return;
+            }
+            _cancellationTokenSource = new CancellationTokenSource();
+            while (!_cancellationTokenSource.IsCancellationRequested)
+            {
+                await BlockUntilNextNight();
+                await ReloadCache();
+            }
+        }
+
+        public void StopGeneratingStatsCacheEveryday()
+        {
+            _cancellationTokenSource.Cancel();
```

Comment:
czemu to sie nazywa `StopGeneratingStatsCacheEveryday` i w środku zatrzymuje cancellation token?

Reply:
ups... kopiuj wklej i nie zauważyłem